docs/howto/enumerations.md

```diff
@@ -0,0 +1,100 @@
+# How to specify enumerations
+
+Enumerations are **finite** sets of values, which may be **named** and may
+have an associated **semantic meaning**.
+
+**Note**
+While enumerations may be used to define [classes](/definitions/classes) or [labels](/definitions/labels), they don't mecessarily share the same meaning, check the definitions.
+
+Although often being of `dataType` [`sc:Text`](http://schema.org/Text), enumerations can also be of other `dataType`s (e.g. `sc:Integer`).
+
+For example, the titanic dataset uses an enumeration to specify
+that the passengers gender field, of type `sc:Text`, can only take two values: `male` or `female`.
+
+## Declaring enumerations
+
+There are two ways to declare enumerations:
+
+1. Directly on a `ml:Field`. This is a simple way to define an enumeration
+   but has some limitations discussed below.
+2. Using a dedicated `ml:RecordSet`, holding the enumeration values. Such
+   definition allows enumerations to be fully defined, with a name and a
+   semantic URL.
+
+### `ml:Field` enumeration
+
+In the following example, extracted from the [coco2014 Croissant definition](https://github.com/mlcommons/croissant/blob/main/datasets/coco2014/metadata.json), the field `supercategory` is declared to be
+an enumeration using the `"ml:isEnumeration": true` property. This means that
+there is a finite number of `supercategory` values, and that it is expected
+that many rows will have the same value.
+
+https://github.com/mlcommons/croissant/blob/6c8eca2925191577a7101dcabd0a880aca03cae3/datasets/coco2014/metadata.json#L259-L271
+
+The field is of `dataType` `sc:Text`, but also has the type `sc:name`. This
+means that `supercategory` is expected to be a text, and that this text is
+the name of the enumeration value   ` naming itself (the
+supercategory), makes sense to humans, and could therefore be used as it in
+UIs.
+
+Although being straightforward, this approach of defining an enumeration has several limitations:
+
+1. The only way to know the cardinality of the enumeration (the number of
```

Comment:
Even then, do we guarantee that all possible values of the enumerations appear in the column?

Reply:
So for enumerations defined as a `ml:Field` enumeration, by definition all the possible values of the enumeration appear in the column (of a given partition).

For enumerations defined as a `ml:RecordSet`, it's a good point: we should specify explicitely what are the guarantees, or say explicitely that there are no guarantees.

I would be in favor of having: "the listed values of the RecordSet enumeration must be a superset of what appears in the joined RecordSet". If there is a need later, we can always add support for "non complete" enumeration RecordSet. 
This allows loading library to know the cardinality of the enumeration, which is important for loading datasets.
What do you think?